aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/UpdateHandler.java

```diff
@@ -102,6 +103,20 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
                     }
                     return progress;
                 })
+                .then(progress -> Commons.execOnce(progress, () -> {
+                    if (isInstanceInStorageFullState(request, rdsClient, progress)) {
+                        if (isAllocatedStorageIncrease(request)) {
+                            return increaseAllocatedStorage(proxy, rdsClient, progress);
+                        }
+                        return ProgressEvent.failed(
+                                request.getDesiredResourceState(),
+                                callbackContext,
+                                HandlerErrorCode.InvalidRequest,
+                                INSTANCE_IS_IN_STORAGE_FULL_STATE_ERROR_MSG
+                        );
```

Comment:
I'm considering letting it fail upon the actual modify-db-instance request in the next step of this chain: thinking of some exotic engine-specific behavior, the factual behavior of an instance in storage-full state is something beyond the public contract, and returning an early failure could limit our flexibility around it. Given we're saving some peanuts by not sending an extra modify-db-instance request, I believe it might be worth it propagating to the API and getting an actual server decision. It will also allow us to get rid of casting an error message locally.